src/AbstractDifferentiation.jl

```diff
@@ -226,44 +226,48 @@ The pushfoward function `pf` accepts as input a `Tuple` of tangents, one for eac
 If `xs` consists of a single element, `pf` can also accept a single tangent instead of a 1-tuple.
 """
 function pushforward_function(ab::AbstractBackend, f, xs...)
-    return (ds) -> begin
-        return jacobian(
-            lowest(ab),
-            (xds...,) -> begin
-                if ds isa Tuple
-                    @assert length(xs) == length(ds)
-                    newxs = xs .+ ds .* xds
-                    return f(newxs...)
-                else
-                    newx = only(xs) + ds * only(xds)
-                    return f(newx)
-                end
-            end,
-            _zero.(xs, ds)...,
-        )
+    function pf(ds)
```

Comment:
Generally I would have preferred to not include such changes in the PR as they are not related to the docstring fixes and make the PR more difficult to review.

Reply:
Generally I would agree but I included this refactoring because it makes this part of the code clearer. I was the one who wrote all the docstrings the first time around, and I screwed up on `value_and_pullback_function` partly because the code was a bit obscure

Reply:
Yeah, I'm fine with it being included in this PR but wanted to mention it nevertheless as a general consideration.

Reply:
Then we're in agreement :)